src/defrag.c

```diff
@@ -60,17 +60,19 @@ int je_get_defrag_hint(void *ptr);
 void *activeDefragAlloc(void *ptr) {
     size_t size;
     void *newptr;
-    if (!je_get_defrag_hint(ptr)) {
+    void *ptr_arr = ptr;
+    defrag_should_defrag_multi(&ptr_arr, 1);
+    if (!ptr_arr) {
         server.stat_active_defrag_misses++;
         return NULL;
     }
     /* move this allocation to a new allocation.
      * make sure not to use the thread cache. so that we don't get back the same
      * pointers we try to free */
     size = zmalloc_size(ptr);
-    newptr = zmalloc_no_tcache(size);
+    newptr = defraged_alloc(size);
```

Comment:
This should be a standard allocation, right? Why do we need `defraged_alloc` macro ? Same question for the free statement below.

Reply:
Today's Malloc and free in defrag are unique to defrag, they skip the tcache, to try and reduce the chance for immediate reallocation of freed memory(this is the only use of these functions). This is totally defrag logic, and tightly depend on the defragging stratedy and therefore I thought it should be in defrag context. I also believe it's not optimal but that is for follow-up discussions